src/main/java/org/eclipse/yasson/internal/serializer/AbstractContainerDeserializer.java

```diff
@@ -120,11 +120,15 @@ protected DeserializerBuilder newUnmarshallerItemBuilder(JsonbContext ctx) {
     }
 
     protected JsonbDeserializer<?> newCollectionOrMapItem(Type valueType, JsonbContext ctx) {
+        //TODO needs performance optimization on not to create deserializer each time
```

Comment:
I'm concerned about this performance regression here. Do we have any immediate need to function as a JPMS module?  If we have to make a tradeoff between performance and JPMS, I'd prefer to stick with performance.

Reply:
Well the performance issue is there regardless of Yasson is loaded as JPMS module or not, I just added a comment, revisiting the code so it is not forgot. In serialization there is already caching of serializer of last item in collection, similar it should be cached for deserialization of collections.

Reply:
added #219 to track it.